### DISCUSSION TITLE
Project panel directory closes when installing project dependencies

### DISCUSSION THREAD

**Discussion by jizhi0v0** [zed-industries/zed#32863] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. open terminal
2. run bun install \
  @blocksuite/presets@0.15.0 \
  @blocksuite/blocks@0.15.0 \
  @blocksuite/store@0.15.0
3. then you can see the left panel automatically shrinks but does not automatically expand

### Environment

Zed: v0.141.3 (Zed)
OS: macOS 14.5.0
Memory: 36 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

https://github.com/zed-industries/zed/assets/57976101/d61ea8d7-fcfb-46b7-b44f-edf0723584a3

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary><pre>

[Zed.log](https://github.com/user-attachments/files/16077608/Zed.log)

</pre></details>

**Comment by osiewicz** [comment]
It looks like this is caused by https://github.com/zed-industries/zed/pull/13072. I wonder if id reuse logic introduced by that PR is not causing us to assign a new node id to the root entry, thus leading project panel to collapse the whole worktree. /cc @mikayla-maki

**Comment by Moosems** [comment]
This has been bugging me so badly, but I don't know Rust well and can't fix it myself with a PR :(

**Comment by Moosems** [comment]
Note that this happens outside the terminal as well (seemingly random in many cases)

**Comment by itpetey** [comment]
I've observed this behaviour whenever the git module reindexes the project, which you can see with trace logging:

```bash
RUST_LOG=worktree=trace /Applications/Zed.app/Contents/MacOS/zed
```

**Comment by JustinLiang90** [comment]
I also noticed this when editing javascript files.

**Comment by JackMatthewRimmer** [comment]
Is there any update on this ? it makes using zed pretty difficult

**Comment by Moosems** [comment]
To be honest it shouldn't shrink on its own unless the user commands such an action, I think the issue should be renamed as it doesn't truly address the problem

**Comment by mattayersnz** [comment]
This issue has been happening to me for a while. It isn't when I'm pressing any keys. I'm on an M1 Mac, and editing JS files.

**Comment by Duclearc** [comment]
I agree that the title isn't great. [I accidentally created a duplicate of this](https://github.com/zed-industries/zed/issues/14488) because I didn't see how this title could have been the same as my problem.
But this is happening so often (many times per day) it makes Zed extremely painful to use.

**Comment by Moosems** [comment]
Fixing this should probably be a higher priority than it is

**Comment by andrebaerlocher** [comment]
I had the same problem, but it stopped occurring. Either because of some kind of fix or because I changed Git-client, currently working with GitButler (which I can really recommend!)

**Comment by raphaellaude** [comment]
I love Zed but this is making the editor completely unusable.

**Comment by itpetey** [comment]
I think I found the culprit (at least on my system). I'm on MacOS and normally develop in `~/Documents/projects`, but by moving to `~/projects`, I don't have this issue anymore.

My theory is that it's to do with iCloud Drive compatibility issues. No idea why, but it'd make sense given that writes to the filesystem seem to trigger the tree collapsing behaviour for me. Hopefully this helps someone else too!

EDIT: To clarify - iCloud Drive synchronises `~/Documents`, but it does not synchronise the whole user home dir. Moving to any location outside of an iCloud Drive synchronised dir should work.

**Comment by Moosems** [comment]
This also happens in the Desktop folder, though moving a project to a root folder isn't ideal by any means

**Comment by itpetey** [comment]
@Moosems Desktop is an iCloud Drive synchronised directory.

<img width="472" alt="Screenshot 2024-07-25 at 11 19 20 AM" src="https://github.com/user-attachments/assets/8b212f1b-8433-4494-b4fd-77cc6159c072">

**Comment by Moosems** [comment]
Right

**Comment by raphaellaude** [comment]
> I think I found the culprit (at least on my system). I'm on MacOS and normally develop in `~/Documents/projects`, but by moving to `~/projects`, I don't have this issue anymore.

Can I give you a big hug 🤗 , @petehayes102 ?! So far so good on my end with the same fix (was in `~/Documents/GitHub`). Thank you! Was driving me crazy

**Comment by chipmusic** [comment]
Can confirm that the issue happens on MacOS when the directory is iCloud-synchronized. To me it happens almost all the time. Simply saving a file with some changes causes it.

**Comment by shellscape** [comment]
@JosephTLyons ping on this one. starting to more widely affect the team.

**Comment by spirosmaggioros** [comment]
> Can confirm that the issue happens on MacOS when the directory is iCloud-synchronized. To me it happens almost all the time. Simply saving a file with some changes causes it.

[osiewicz](https://github.com/osiewicz) I have the same problem as well, the team should prioritize fixing this issue because it is really annoying for macos users. I love the simplicity and customization and im using it as my main text editor right now(instead of neovim), but i only have this issue and it's not minor at all, the view panel randomly just collapses all the time.
We can't just remove the projects from icloud because of an editor

Edit: I removed all my github projects from icloud and the issue is fixed, but still this should be fixed.

**Comment by matthieuHenocque** [comment]
This issue has been made worse with the 0.147.2 build

Now the root folder will collapse randomly when editing a file

**Comment by joshuaia** [comment]
I have this problem too, it makes using Zed unusable. Moving working directory to the different location doesn't help.

Does anybody still have this issue? It looks like to be a critical issue, but the last message is 2 weeks old.
I'm wondering how did you manage to find a workaround.

**Comment by maxbrunsfeld** [comment]
I believe this is fixed in 
- https://github.com/zed-industries/zed/pull/16926

**Comment by shellscape** [comment]
thanks for your work getting this resolved

**Comment by mattayersnz** [comment]
I've had this issue for a while now, working with js files, all I need to
do is update the file and save it, and the project tree collapses. There
has been a lot of different discussions about related issues, but it still
hasn't been resolved. I've moved back to another editor now, but hope it
gets fixed in the future.

**Comment by matthieuHenocque** [comment]
The bug is still there.

To reproduce:
- fresh install Zed
- open a project folder and open the project panel
- open any javascript file (mine is in a folder named `modules` at the root of the project)
- make some edits

I can't work that way.

**Comment by osiewicz** [comment]
@matthieuHenocque unless you're building Zed from source, this fix will be available starting with tomorrows Preview build (and next weeks Stable)

**Comment by SnapshotCiTy** [comment]
still happening to me with Zed 0.152.3 on Sonoma 14.6.1 - iCloud drive synchronised Documents folder and projects inside of ~/Documents/something/some_project - Go projects and Hugo projects (but that seem irrelevant)...

**Comment by JosephTLyons** [comment]
Just had a friend message me and say they saw this bug again the other day - they think it is a regression. Is anyone else seeing this bug again?

**Comment by itpetey** [comment]
@JosephTLyons No I can't replicate on latest stable (0.155.2).

**Comment by shellscape** [comment]
I can still replicate this issue on subdirectories on the latest preview build (as of writing this). @JosephTLyons I'll DM you a video of this happening, as I can't share the repo contents publicly.

**Comment by itpetey** [comment]
It's probably nothing but I did upgrade to MacOS 15 in the intervening period between tests. Might be worth considering if you're drawing a blank.

**Comment by shellscape** [comment]
fwiw I have no upgraded MacOS. Still on Ventura 13.3.1 (a) . @JosephTLyons video was sent in X DM on Oct 2nd.

**Comment by JosephTLyons** [comment]
A video of this happening on a private repo can be found in our company Slack [here](https://zed-industries.slack.com/archives/C04SYPRCKPF/p1736921809131039).

**Comment by smitbarmase** [comment]
I'm not able to reproduce it with https://github.com/shellscape/jsx-email on v0.182.0, wonder if it's macOS thing or if we changed anything since Oct. I'm on macOS 15.4. 

https://github.com/user-attachments/assets/62ebaccf-8771-49d8-8f05-815f71a0cc4b

**Comment by shellscape** [comment]
@smitbarmase I'm on 15.3 using the latest preview version, and I'm opted into the beta. Can confirm the repro above also does not collapse for me. When I followed up with @JosephTLyons the other day, it was because I was working in another repo where this was happening consistently. It's not a repo I can share publicly, but I'll get Joseph another video when I'm back in that repo. If you're on the discord channel I can send it directly to you as well. There must be a specific set of conditions that leads to the behavior - I'll figure that out.

**Comment by smitbarmase** [comment]
It will be helpful to hunt this down if we can reproduce it on any public repo. I'm on Discord, but I don't think video will help since I've already seen what's happening. Let me know if it happens for you on any public repo, that would be best.

At least now we know it's not macOS, but very certain type of case. I now suspect if it's somehow related to git, as all the repro videos I saw had collapsed directory with git indicator.

**Comment by shellscape** [comment]
Disappointed this was dismissed given the effort put into reporting and triage and the fact it's still a continuing and valid issue.

**Comment by SomeoneToIgnore** [comment]
For this and other continuing and valid issues, it's also somewhat disappointing to get no repro steps or details for over 2 month, hence I've decided to close a few of such.

I do not really dismiss any of these including the current one, but trying to move out unactionable ones out of the issue tracker.

The ones with non-zero activity are worth moving into discussions, sure.

**Reply by shellscape** [reply]
Holy needless snark, Batman. Is that the modus operandi of the team now?

My dude I have volunteered my time _several times_ to join calls to triage live on a repo I cannot grant access to. None of those offers have been accepted. Your reply is insulting.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
